vyper/context/validation/data_positions.py

```diff
@@ -0,0 +1,40 @@
+import math
+
+from vyper import ast as vy_ast
+from vyper.context.types.bases import StorageSlot
+
+
+def set_data_positions(vyper_module: vy_ast.Module) -> None:
+    """
+    Parse the annotated Vyper AST, determine data positions for all variables,
+    and annotate the AST nodes with the position data.
+
+    Arguments
+    ---------
+    vyper_module : vy_ast.Module
+        Top-level Vyper AST node that has already been annotated with type data.
+    """
+    set_storage_slots(vyper_module)
+
+    # for fn_node in vyper_module.get_children(vy_ast.FunctionDef):
+    #     set_calldata_offsets(fn_node)
+    #     set_memory_offsets(fn_node)
```

Comment:
Needed?

Reply:
No, just showing the shape for the future.  Can remove now - it'll be in another PR shortly after this merges.